lib/active_model/serializer/reflection.rb

```diff
@@ -17,7 +25,28 @@ class Serializer
     #
     # So you can inspect reflections in your Adapters.
     #
-    Reflection = Struct.new(:name, :options) do
+    Reflection = Struct.new(:name, :options, :block) do
+      delegate :call, to: :reader
```

Comment:
I just think it reads better. It might be a performance issue.

I think there's definitely room for refactoring and revising this, but I don't think it should be a blocker.  I don't really like how the reflection differs from the attribute, but figure we can kick that can down the road a bit more into its own PR.